test/functional-tests/ACTCampaignEngine.py

```diff
@@ -0,0 +1,94 @@
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+# this list of conditions and the following disclaimer in the documentation and/or
+# other materials provided with the distribution.
+#
+# 3. Neither the name of the copyright holder nor the names of its contributors
+# may be used to endorse or promote products derived from this software without
+# specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+# ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""
+Create a test suite for all tests about SET/GET commands
+
+Set the test environment.
+
+Uses PfwSetTsetSuite to create a single instance of the HAL
+for all the SET/GEt commands.
+
+These commands are tested using the methods of the classes
+"BooleanTestCase", etc...
+"""
+
+import os, sys
+import unittest
+import shutil
+from Util import PfwUnitTestLib
+
+
+class Logger(object):
+
+    def __init__(self, filename="Default.log"):
+        self.terminal = sys.stdout
+        self.log = open(filename, "a")
+
+    def write(self, message):
+        self.terminal.write(message)
+        self.log.write(message)
+
+def testsRunner(testDirectory):
+
+    tests = unittest.defaultTestLoader.discover(testDirectory, pattern='t*.py')
+    runner = unittest.TextTestRunner(verbosity=2)
+
+    return runner.run(tests).wasSuccessful()
+
+def main():
+
+    os.environ["PFW_FILESYSTEM"] = os.environ["PFW_FS"]
+    os.environ["PFW_TEST_TOOLS"] = "."
+    os.environ["PFW_TEST_CONFIGURATION"] = os.environ["PFW_FS"] + "xml/configuration/ParameterFrameworkConfiguration.xml"
+
+    isAlive = os.environ["PFW_FS"] + "isAlive"
+    with open(isAlive, 'w') as fout:
+        fout.write('true')
+
+    needResync = os.environ["PFW_FS"] + "needResync"
+    with open(needResync, 'w') as fout:
+            fout.write('false')
+
+    res = testsRunner('PfwTestCase')
+#    shutil.rmtree(os.environ["PFW_FS"])
+    for path, dirs, files in os.walk(  os.environ["PFW_FS"] ):
+        if 'xml' in dirs:
+            dirs.remove('xml')
+        for name in files:
+            os.remove(os.path.join(path, name))
+        for name in dirs:
+            os.rmdir(os.path.join(path, name))
+        
+    sys.exit(not res)
+
+if __name__ == "__main__":
+    main()
```

Comment:
1. When creating file paths, use os.path.join instead of "+" betwteen strings, which may be wrong (e.g. "/some/path" + "someFileName" == /some/pathsomeFileName" whereas os.path.join("/some/path", "someFileName") == "/some/path/someFileName")
2. remove the commented line
3. remove surrounding spaces around ˋos.environ["PFW_FS"]ˋ